migrator/tests/streams/actions/test_draft_actions_stream.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# ZenodoRDM is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Test draft actions stream."""
+
+from uuid import UUID
+
+import pytest
+import sqlalchemy as sa
+from invenio_rdm_migrator.extract import Extract
+from invenio_rdm_migrator.load.postgresql.transactions import PostgreSQLTx
+from invenio_rdm_migrator.streams import Stream
+from invenio_rdm_migrator.streams.models.files import FilesBucket
+from invenio_rdm_migrator.streams.models.pids import PersistentIdentifier
+from invenio_rdm_migrator.streams.models.records import (
+    RDMDraftMetadata,
+    RDMParentMetadata,
+    RDMVersionState,
+)
+from invenio_rdm_migrator.transform import Tx
+
+from zenodo_rdm_migrator.actions import ZenodoDraftCreateAction
+
+
+@pytest.fixture()
+def test_extract_cls(
+    create_draft_tx,
+    parents_state,
+    records_state,
+    communities_state,
+    pids_state,
+    global_state,  # required for PID values management
+):
+    class TestExtractor(Extract):
+        """Test extractor.
+
+        Skips the transaction grouping and fingerprinting steps.
+        """
+
+        def run(self):
+            """Yield one element at a time."""
+            yield ZenodoDraftCreateAction(
+                tx_id=create_draft_tx["tx_id"],
+                data=create_draft_tx["operations"],
+                parents_state=parents_state,
+                records_state=records_state,
+                communities_state=communities_state,
+                pids_state=pids_state,
+            )
+
+    return TestExtractor
+
+
+DB_URI = "postgresql://zenodo:zenodo@localhost:5432/zenodo"
+
+
+@pytest.fixture(scope="function")
+def db_engine():
+    tables = [
+        PersistentIdentifier,
+        FilesBucket,
+        RDMDraftMetadata,
+        RDMParentMetadata,
+        RDMVersionState,
+    ]
+    eng = sa.create_engine(DB_URI)
+
+    # create tables
+    for model in tables:
+        model.__table__.create(bind=eng, checkfirst=True)
+
+    yield eng
+
+    # remove tables
+    for model in tables:
+        model.__table__.drop(eng)
+
+
+def test_draft_create_action_stream(test_extract_cls, db_engine):
+    """Creates a DB on disk and initializes all the migrator related tables on it."""
+    # test
+
+    stream = Stream(
+        name="action",
+        extract=test_extract_cls(),
+        transform=Tx(),
+        load=PostgreSQLTx(DB_URI),
+    )
+    stream.run()
+
+    with db_engine.connect() as conn:
```

Comment:
Did not use Core because there was an issue with the declarative classes and also to distance it from the modeling used by the migrator. This way the queries are more explicit on what we check.